src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1.cs

```diff
@@ -548,22 +548,17 @@ public ImmutableArray<T> InsertRange(int index, IEnumerable<T> items)
             {
                 return ImmutableArray.CreateRange(items);
             }
-
-            int count = ImmutableExtensions.GetCount(ref items);
-            if (count == 0)
+            
+            T[] other = items.ToArray();
```

Comment:
@sharwell I have fixed this in a newer commit, at the cost of expanding the code by about 10 lines in a check for `ICollection<T>`. There is no longer a buffer between `items` and the dest array.